File: src/backend/facilitymanagement/src/main/java/org/taonaw/facilitymanagement/domain/model/equipmentcategory/EquipmentCategoryName.java
package org.taonaw.facilitymanagement.domain.model.equipmentcategory;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import org.taonaw.facilitymanagement.domain.shared.Assertion;

@Getter
@EqualsAndHashCode
public class EquipmentCategoryName {
    private final String value;

    public EquipmentCategoryName(@NonNull String value) {
        Assertion.argumentRange(value, 1, 50);
        this.value = value;
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/equipmentCategory/EquipmentCategoryName.kt
package org.taonaw.facility.domain.model.equipmentCategory

data class EquipmentCategoryName(val value: String) {

    companion object {
        const val LENGTH_MIN = 1
        const val LENGTH_MAX = 20
    }

    init {
        require(value.length in LENGTH_MIN..LENGTH_MAX) {
            "機材カテゴリ名は $LENGTH_MIN 文字以上 $LENGTH_MAX 文字いかでなければなりません。"
        }
    }
}